### PR TITLE
Revert Email config changes

### PR DESCRIPTION
**Pull request by maestromac**
## What type of PR is this? (check all applicable)
- [x] Bug Fix

## Description
The App isn't pull proper configuration.

### CODE REVIEW

File: app/mailers/application_mailer.rb [hunk 1]
```diff
@@ -8,8 +8,10 @@ class ApplicationMailer < ActionMailer::Base
   helper AuthenticationHelper
 
   before_action :use_custom_host
-  before_action :set_delivery_options
-  before_action :set_perform_deliveries
+
+  # [@SRE] temporarily disabled
+  # before_action :set_delivery_options
```
Comment (djuber) [inline app/mailers/application_mailer.rb:13]: These are commented out in favor of the (reintroduced) settings from production.rb

File: config/environments/production.rb [hunk 2]
```diff
@@ -127,8 +127,27 @@
   protocol = ENV["APP_PROTOCOL"] || "http://"
 
   config.action_mailer.delivery_method = :smtp
-  config.action_mailer.perform_deliveries = false
+  config.action_mailer.perform_deliveries = true
```
Comment (djuber) [inline config/environments/production.rb:130]: this reverts changes from the PR adding Settings::SMTP

File: spec/system/user_logs_in_with_password_spec.rb [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 RSpec.describe "Authenticating with a password" do
   def submit_login_form(email, password)
+    allow(ForemInstance).to receive(:smtp_enabled?).and_return(true)
```
Comment (djuber) [inline spec/system/user_logs_in_with_password_spec.rb:5]: this fixes a failing spec (not sure why it passed in the PR?) - oddly this works when run in isolation and may be tied to interaction (see the clear_cache change earlier).

File: spec/models/settings/smtp_spec.rb [hunk 4]
```diff
@@ -1,6 +1,10 @@
 require "rails_helper"
 
 RSpec.describe Settings::SMTP do
+  after do
+    described_class.clear_cache
```
Comment (djuber) [inline spec/models/settings/smtp_spec.rb:5]: this is new and solves another related problem?

**Review by djuber — APPROVED** [review]
Good luck

File: spec/models/settings/smtp_spec.rb (hunk 4 above)
Comment (maestromac) [reply spec/models/settings/smtp_spec.rb:5]: @djuber yep the settings here were bleeding into another spec.

**Review by Zhao-Andy — APPROVED** [review]

**Pull request merged**